src/infra/core/src/pull_service_impl.rs

```diff
@@ -431,6 +444,16 @@ impl PullServiceImpl {
             })
             .collect())
     }
+
+    fn classify_dataset_reference(
+        &self,
+        r: &DatasetRefAny,
+    ) -> Result<DatasetRef, DatasetRefRemote> {
+        // Single-tenant workspace => treat all repo-like references as repos.
+        // Multi-tenant workspace => treat all repo-like references as accounts, use
+        // repo:// for repos
```

Comment:
Does `repo//name` format actually work? I didn't see any changes in the reference parsing code.

Reply:
Never tried. We need to extend tests to check